src/main/java/sg/ncl/testbed_interface/Dataset.java

```diff
@@ -1,155 +1,54 @@
 package sg.ncl.testbed_interface;
 
-import java.util.HashSet;
-import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
 
-public class Dataset {
-	
-	
-	private String datasetName;
-	private String releaseDate;
-	private String datasetDescription;
-	private String license;
-	private int ownerId;
-	private int datasetId;
-	private String isPublic = "public"; // dataset visibility: private and public
-	private boolean requireAuthorization; // true: need to request access Restricted, false: can just download Open
-	private Set<Integer> usersAccessSet = new HashSet<Integer>(); // stores a list of userids that have accessed to this dataset including the owner
-	private boolean hasAcceptDataOwnerPolicy;
-	private String fileName;
-	
-	public Dataset() {
-	}
-
-	public String getDatasetName() {
-		return datasetName;
-	}
-
-	public void setDatasetName(String datasetName) {
-		this.datasetName = datasetName;
-	}
-
-	public String getReleaseDate() {
-		return releaseDate;
-	}
-
-	public void setReleaseDate(String releaseDate) {
-		this.releaseDate = releaseDate;
-	}
-
-	public String getIsPublic() {
-		return isPublic;
-	}
-
-	public void setIsPublic(String isPublic) {
-		this.isPublic = isPublic;
-	}
-
-	public int getOwnerId() {
-		return ownerId;
-	}
-
-	public void setOwnerId(int ownerId) {
-		this.ownerId = ownerId;
-	}
-
-	public String getDatasetDescription() {
-		return datasetDescription;
-	}
-
-	public void setDatasetDescription(String datasetDescription) {
-		this.datasetDescription = datasetDescription;
-	}
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
 
-	public String getLicense() {
-		return license;
-	}
+@Getter
+@Setter
+public class Dataset implements Serializable {
 
-	public void setLicense(String license) {
-		this.license = license;
-	}
+	private Integer id;
+	private String name;
+	private String description;
+	private String contributorId;
+	private String visibility;
+	private String accessibility;
+	private List<DataResource> dataResources;
+	private List<String> approvedUsers;
 
-	public int getDatasetId() {
-		return datasetId;
-	}
+	private User2 contributor;
+	
+	public Dataset() {
+	    dataResources = new ArrayList<>();
+	    approvedUsers = new ArrayList<>();
+    }
 
-	public void setDatasetId(int datasetId) {
-		this.datasetId = datasetId;
-	}
+	public boolean isOpen() {
+	    return accessibility.equalsIgnoreCase("open");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the "open" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)